lib/task-graph.js

```diff
@@ -61,13 +61,57 @@ function taskGraphFactory(
         };
         if (this.context.target) {
             this.logContext.id = this.context.target;
-        }
+       }
         // For database ref linking
         this.node = this.context.target || this.definition.options.nodeId;
 
         return this;
     }
 
+    /**
+    This is a helper that visits one node. 
+    This function is used in _visitGraphNode().
+    */
+
+    TaskGraph.prototype._visitOneGraphNode = function(task, value, dep) {
```

Comment:
This is  the helper function that helps interating in function _visitGraphNode(). By comaring the original code, this function is changed by adding a conditional check and iterate. Same with _processWaitOnTask().